Replicas: Improve sorting replicas; #4017

bziemons opened the pull request:
Improve sorting replicas

Deduplicate code related to replica sorting
Set geoip as default sorting option for all
Return lan locations always before wan locations for list_replicas
Replace tests for geoip replica sorting
Fix setting priority to the order of what the sort returns on list_replicas endpoint

bziemons commented:
Depends on #4081

mlassnig commented on lib/rucio/web/rest/webpy/v1/redirect.py:
```diff
@@ -142,17 +143,7 @@ def GET(self, scope, name):
-                    replicas = sort_ranking(dictreplica, client_location)
-                else:
-                    replicas = sort_random(dictreplica)
+                replicas = sort_replicas(dictreplica, client_location, selection=select, default='random')
```
should this be default=geoip instead?

bziemons replied:
no, the redirect endpoint is out of scope.

mlassnig commented on lib/rucio/web/rest/webpy/v1/redirect.py:
```diff
@@ -277,17 +268,7 @@ def GET(self, scope, name):
-                            else:
-                                rep = sort_random(dictreplica)
-
+                            rep = sort_replicas(dictreplica, client_location, selection=select, default='random')
```
same here

bziemons replied:
no, the redirect endpoint is out of scope.

mlassnig commented on lib/rucio/web/rest/flaskapi/v1/redirect.py:
```diff
@@ -141,17 +142,7 @@ def generate():
-                        replicas = sort_ranking(dictreplica, client_location)
-                    else:
-                        replicas = sort_random(dictreplica)
+                    replicas = sort_replicas(dictreplica, client_location, selection=sortby, default='random')
```
default=geoip?

bziemons replied:
no, the redirect endpoint is out of scope.

mlassnig commented on lib/rucio/web/rest/flaskapi/v1/redirect.py:
```diff
@@ -283,17 +274,7 @@ def get(self, scope_name):
-                            else:
-                                rep = sort_random(dictreplica)
-
+                            rep = sort_replicas(dictreplica, client_location, selection=select, default='random')
```
same here?

bziemons replied:
no, the redirect endpoint is out of scope.

bziemons committed:
Replicas: Improve sorting replicas; Fix #4017

Deduplicate code related to replica sorting
Set geoip as default sorting option for all
Return lan locations always before wan locations for list_replicas
Replace tests for geoip replica sorting
Fix setting priority to the order of what the sort returns on list_replicas endpoint
